vision/imageloader/imageloader.go

```diff
@@ -73,10 +82,22 @@ func New(fn string, vocab map[string]int, trans *transforms.ComposeTransformer,
 		bufSize:     bufSize,
 		seed:        seed,
 	}
-	runtime.LockOSThread()
-	go m.readSamples()
 	go m.shuffleSamples()
-	go m.samplesToMinibatches()
+	// `readSamples` and `samplesToMinibatches` calls `gocv`, we make them only run in background OS threads.
+	readSamplesCh <- func() { m.readSamples() }
+	samplesToMinibatchesCh <- func() { m.samplesToMinibatches() }
+	counterMu.Lock()
```

Comment:
It seems that the mutex makes `imageLoader` can be created in the different goroutines, am I right? Do you want to add a unit test about the mutext.

Reply:
No. We recommend `ImageLoader`s only used in the main goroutine.